Source/OxyPlot.SkiaSharp.Avalonia/DoubleBuffered/PlotRenderer.cs

```diff
@@ -0,0 +1,225 @@
+                    {
+                        lock (plotModel.SyncRoot)
+                        {
+                            this.Render(iPlotModel, size);
+                        }
+                    }
+                }
+            }
+        }
+
+        private void Render(IPlotModel model, Size size)
+        {
+            var scale = this.EnsureBackBuffer(size);
+
+            using (var canvas = new SKCanvas(this.backBuffer))
+            {
+                canvas.Scale((float)scale);
+                this.renderContext.SkCanvas = canvas;
+                canvas.Clear(model.Background.ToSKColor());
+                model.Render(this.renderContext, new OxyRect(0, 0, this.backBuffer.Width / scale, this.backBuffer.Height / scale));
+            }
+
+            this.BufferSwitch();
+            Dispatcher.UIThread.InvokeAsync(this.InvalidateVisual);
+        }
+
+        /// <inheritdoc />
+        protected override void OnAttachedToVisualTree(VisualTreeAttachmentEventArgs e)
+        {
+            base.OnAttachedToVisualTree(e);
+            this.StartRenderLoop();
+        }
+
+        /// <inheritdoc />
+        protected override void OnDetachedFromVisualTree(VisualTreeAttachmentEventArgs e)
+        {
+            base.OnDetachedFromVisualTree(e);
+            this.StopRenderLoop();
+        }
+
+        private void StartRenderLoop()
+        {
+            new Task(() =>
+            {
+                this.renderLoopMutex.WaitOne();
+                this.renderException = null;
+                this.renderCancellationTokenSource = new CancellationTokenSource();
+
+                try
+                {
+                    this.RenderLoop(this.renderCancellationTokenSource.Token);
+                }
+                catch (OperationCanceledException)
+                {
+                }
+                catch (Exception e)
+                {
+                    this.renderException = e;
+                }
+                finally
+                {
+                    this.backBuffer?.Dispose();
+                    this.frontBuffer?.Dispose();
+                    this.frontBuffer = this.backBuffer = null;
+                    this.renderLoopMutex.ReleaseMutex();
+                }
+            }, TaskCreationOptions.LongRunning).Start();
+        }
+
+        private void StopRenderLoop()
+        {
+            this.renderCancellationTokenSource?.Cancel();
+            this.renderCancellationTokenSource?.Dispose();
+            this.renderCancellationTokenSource = null;
+        }
+
+        /// <inheritdoc />
+        public void Dispose()
+        {
+            // First stop the render loop
+            this.StopRenderLoop();
+
+            // release renderRequiredEvent once more in case the render loop currently is waiting for the mutex; otherwise it might wait forever and will not be canceled
+            this.RequestRender();
```

Comment:
If we can switch to using `async` for the render loop, can then await the cancellation token rather than having to pulse it when we are disposing. As is, I think this should be in `StopRenderLoop` so that `OnDetachedFromVisualTree` also ensures the render loop doesn't sit waiting.